test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2550,8 +2551,16 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
             param.flags         = flags;
 
             ucs_status_ptr_t sreq = ucp_am_send_nbx(sender().ep(), 0,
-                                                    &shdr[0], hdr_size,
+                                                    &shdr_copy[0], hdr_size,
                                                     &sb[0], size, &param);
+            /* First message request triggers connection establishment and
+             * is placed into pending queue.
+             * To check UCP_AM_SEND_FLAG_COPY_HEADER we change AM header 
+             * content while the request is still in pending queue.*/ 
+            if (flags & UCP_AM_SEND_FLAG_COPY_HEADER) {
+                shdr_copy[0] = 'X';
```

Comment:
1. the shdr_copy should remain the original value and shdr should be "corrupted"
2. i would fill entire "corrupted" header with aaa..